src/nodes/hypertable_modify.c

```diff
@@ -983,7 +983,8 @@ ExecModifyTable(CustomScanState *cs_node, PlanState *pstate)
 				if (unlikely(!resultRelInfo->ri_projectNewInfoValid))
 					ExecInitInsertProjection(node, resultRelInfo);
 				slot = ExecGetInsertNewTuple(resultRelInfo, context.planSlot);
-				slot = ExecInsert(&context, cds->rri, slot, node->canSetTag);
+				Assert(cds);
+				slot = ExecInsert(&context, (cds ? cds->rri : NULL), slot, node->canSetTag);
```

Comment:
I guess you can avoid this change since you've added an assert in the previous line, or am I missing something?

Reply:
Not sure it is valid to pass in NULL to `ExecInsert`... checking...

Yup, this is eventually passed down to `ExecFindPartition`, which tries to de-reference it, so you should not pass in NULL for this parameter to `ExecInsert`.